src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlDuplicateSettersChecker.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using XamlX;
+using XamlX.Ast;
+using XamlX.Transform;
+
+namespace Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers;
+
+class AvaloniaXamlIlDuplicateSettersChecker : IXamlAstTransformer
+{
+    public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode node)
+    {
+        if (node is not XamlAstObjectNode objectNode)
+        {
+            return node;
+        }
+
+        var fullName = objectNode.Type.GetClrType().FullName;
+        if (fullName is not ("Avalonia.Styling.Style" or "Avalonia.Styling.ControlTheme"))
```

Comment:
From the `context.GetAvaloniaType()` you can get an object with well-known types. I believe it should include Style and ControlTheme (if not - it can be added).
Having that, you can more reliably check types with IsAssignableFrom.

As an example:
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlDataContextTypeTransformer.cs#L103

Reply:
I took the examples from here

https://github.com/AvaloniaUI/Avalonia/blob/c59466d389eb87168c3ed326b50b613721295a53/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlControlThemeTransformer.cs#L14

https://github.com/AvaloniaUI/Avalonia/blob/c59466d389eb87168c3ed326b50b613721295a53/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs#L22

Should I modify these checks too?

Reply:
At least GetAvaloniaTypes can be used there, yes